actionpack/lib/system_testing/test_helpers/screenshot_helper.rb

```diff
@@ -0,0 +1,47 @@
+module SystemTesting
+  module TestHelpers
+    # Screenshot helper for system testing
+    module ScreenshotHelper
+      # Takes a screenshot of the current page in the browser if the system
+      # test driver supports screenshots and the test failed.
+      #
+      # Additionally +take_screenshot+ can be used within your tests at points
+      # you want to take a screenshot if the driver supports screenshots. The
+      # Rack Test driver does not support screenshots.
+      #
+      # You can check of the driver supports screenshots by running
+      #
+      #   Rails::SystemTestCase.driver.supports_screenshots?
+      #   => true
+      def take_screenshot
+        puts "[Screenshot]: #{image_path}"
+        puts find_image
+      end
+
+      private
+        def supported?
+          Rails::SystemTestCase.driver.supports_screenshots? && !passed?
+        end
+
+        def image_path
+          path = "tmp/screenshots/failures_#{method_name}.png"
+          page.save_screenshot(Rails.root.join(path))
+          path
```

Comment:
Find it a little odd that `image_path` both returns a path and saves an image. I mean, wouldn't the two calls to `image_path` in `find_image` save two more images?